manifests/candlepin.pp

```diff
@@ -130,6 +121,30 @@
       key_mode    => '0440',
     }
 
+    file { $keystore_password_path:
+      ensure  => file,
+      content => $keystore_password,
+      owner   => 'root',
+      group   => $group,
+      mode    => '0440',
+    }
+
+    keystore { $keystore:
+      ensure        => present,
+      password_file => $keystore_password_path,
+      owner         => 'root',
+      group         => $group,
+      mode          => '0640',
+    }
+
+    keystore_certificate { "${keystore}:tomcat":
```

Comment:
I must admit I find the notion of `$keystore:$alias` for the name rather confusing. Is this a common notation in the Java/keytool universe?

I see you could also write
```
keystore_certificate { "funny_name":
  keystore => $keystore,
  alias     => 'tomcat'
  …
}
```

And I find this more readable. But wouldn't block this on my weird brain :)

Reply:
It's more of a Puppet convention. I think that what you write will also work with the current code. The benefit of the current version is that you can write `tomcat` into multiple keystores without having duplicate resources, so this works:
```puppet
keystore_certificate { "${keystore1}:tomcat":
}
keystore_certificate { "${keystore2}:tomcat":
}
```

If you don't, you need to use things like `funny_name1` and `funny_name2`. There are arguments for both.

That said, it's probably good to write a test case for both syntaxes to verify it really does work for both versions.

Reply:
Yeah, I also read the code that both notations *should* work.

Reply:
I tested this, but:

```
  Error: Evaluation Error: Error while evaluating a Resource Statement, No set of title patterns matched the title "funny_name". (file: /etc/puppetlabs/code/environments/production/modules/certs/manifests/candlepin.pp, line: 140, column: 5) on node centos7-64-1
```

My code was:

```puppet
    keystore_certificate { "funny_name":
      keystore      => $keystore,
      alias         => 'tomcat',
      ensure        => present,
      password_file => $keystore_password_path,
      certificate   => $tomcat_cert,
      private_key   => $tomcat_key,
      ca            => $ca_cert,
    }
```

Reply:
I've updated this to support passing:

  1) either just `keystore` in which case the name is the `alias`
  2) `keystore:alias`
  3) `keystore` and `alias` parameters with a random name

My workflow is adding some stand alone provider tests in follow up PRs so I will handle testing some of that in there.

Reply:
Can we mark this as resolved now?